support to input module to handle input paramters string

anto6715 opened the pull request:
@mdrudi Here the pull request related to the introduction of the module to handle the **input_parameters** string.

I already integrated the module and prepared a first working prototype of the mockup.
At the moment the documentation is missing, I'll prepare a little README to explain how the module works

anto6715 committed:
initial support to input module to handle input paramters string; update README with the new command using input_paramters string

mdrudi commented:
Also to consider a readme for the documentation of the input string syntax, folowing what was done in https://docs.google.com/document/d/12nQMglTqrDqxGOJzD2nUBfkiUDiFvML9EWT0SynLi4o/edit#heading=h.v1qeahi65eh8

anto6715 committed:
Add README for input module

anto6715 commented:
> Also to consider a readme for the documentation of the input string syntax, folowing what was done in https://docs.google.com/document/d/12nQMglTqrDqxGOJzD2nUBfkiUDiFvML9EWT0SynLi4o/edit#heading=h.v1qeahi65eh8

Here a first version of the README: https://github.com/CMCC-Foundation/bluecloud-mockup_method/tree/dev_input_parameters/input

mdrudi commented:
What are the commonon parameters between all methods ?

anto6715 commented:
> Concerning KNMI :
> 
>     *

I update the document to add these new parameters, if it is ok I'll proceed to update also the README.
Here the link to the document:

https://docs.google.com/document/d/12nQMglTqrDqxGOJzD2nUBfkiUDiFvML9EWT0SynLi4o/edit#

anto6715 commented:
> * [x]  What are the commonon parameters between all methods ?
> 
>     * [x]  example of specialization
> 
>     * [x]  to check the use of class, parameters,...
> 

README of input module updated

anto6715 committed:
move _raise_type_exception in IParameters class in order to be inherited
add an example how to extend the IParameter class
add list of the common attributes

anto6715 committed:
Add support to different output type in mockup method

anto6715 committed:
Update README adding new two parameter_class

mdrudi commented:
In ordine logico, credo che oltre alle informazioni sul modulo, il reame dovrebbe indicare  : 
- [A] la tipologia di parametri previsti e gestibili
- [B] la specifica della descrizione dei parametri, che puo' essere utilizzata ad esempio per istruire l'UI, ed in generale puo' essere visto come una informazione disponibile "staticamente" non necessariamente generata on-the-fly da un register
- la specifica della corrispondente stringa da generare e mandare al metodo

L'attuale paragrafo Parameters, forse sarebbe meglio scorporarlo in due parti, una focalizzata sulla tipologia di prametri previsti [A], ed una maggiormaente focalizzata su come la UI ricevera' informazioni sui parametri da chiedere all'utente [B].

anto6715 commented:
> L'attuale paragrafo Parameters, forse sarebbe meglio scorporarlo in due parti, una focalizzata sulla tipologia di prametri previsti [A], ed una maggiormaente focalizzata su come la UI ricevera' informazioni sui parametri da chiedere all'utente [B].

Sostanzialmente il punto [A] sarebbe l'attuale tabella giusto?
Resta invece da definire da 0 il punto [B]

anto6715 commented:
@mdrudi A first draft of the document is ready here:  https://docs.google.com/document/d/1GwhsSIZMQFPfhzkMHhyIpAVjoKv9GFenCB0Rb2c3EQA/edit#

anto6715 committed:
now only data source and output type are mandatory parameters for all methods; add working domain missing validation

anto6715 committed:
update README in input module

anto6715 commented:
@mdrudi I update the README aligning it with the drive document.
I also removed the section regarding the parameter class because both method and module are unaware about this information can be a little confusing here.

I think that should be better to add a reference to the drive document when it will be completed. If for you is ok, I can proceed to close this issue. Other correction to the REDME can be made directly in fix_v2 branch

anto6715 committed:
add mandatory note for data_source and id_output_type parameters
